sound/soc/sof/topology.c

```diff
@@ -1817,19 +1817,17 @@ static int sof_widget_load_mux(struct snd_soc_component *scomp, int index,
 	struct snd_sof_dev *sdev = snd_soc_component_get_drvdata(scomp);
 	struct snd_soc_tplg_private *private = &tw->priv;
 	struct sof_ipc_comp_mux *mux;
+	size_t ipc_size = sizeof(*mux);
 	int ret;
 
-	mux = kzalloc(sizeof(*mux), GFP_KERNEL);
+	mux = (struct sof_ipc_comp_mux *)
+	      sof_comp_alloc(swidget, &ipc_size, index, core);
+	mux = kzalloc(ipc_size, GFP_KERNEL);
 	if (!mux)
 		return -ENOMEM;
 
 	/* configure mux IPC message */
```

Comment:
@keyonjie one thing i don't get here. Maybe the diff on my side is wrong. 
``
mux = (struct sof_ipc_comp_mux *)
	      sof_comp_alloc(swidget, &ipc_size, index, core);
mux = kzalloc(ipc_size, GFP_KERNEL);
```

Why do we need the kzalloc call?

Reply:
> @keyonjie one thing i don't get here. Maybe the diff on my side is wrong.
> 
> ```
> + mux = (struct sof_ipc_comp_mux *)
> 	      sof_comp_alloc(swidget, &ipc_size, index, core);
> + mux = kzalloc(ipc_size, GFP_KERNEL);
> ```
> 
> Why do we need the kzalloc call?

Good catch, it is copy/paste error when switching from kzalloc to sof_comp_alloc(), the reason why it didn't lead to any error is that no topology has used this soc_dapm_mux/demux type yet. Thank you and updating it now...